src/battle-log.ts

```diff
@@ -93,16 +99,37 @@ class BattleLog {
 			}
 			break;
 
-		case 'join': case 'j': {
+		case 'join': case 'j': case 'leave': case 'l': {
 			const user = BattleTextParser.parseNameParts(args[1]);
-			divHTML = '<small>' + BattleLog.escapeHTML(user.group + user.name) + ' joined.</small>';
-			break;
-		}
-		case 'leave': case 'l': {
-			const user = BattleTextParser.parseNameParts(args[1]);
-			divHTML = '<small>' + BattleLog.escapeHTML(user.group + user.name) + ' left.</small>';
+			const formattedUser = BattleLog.escapeHTML(user.group + user.name);
+			const isJoin = args[0].includes('j');
+			if (!this.joinLeave) {
+				this.joinLeave = {
+					joins: [],
+					leaves: [],
+					element: document.createElement('div') as HTMLDivElement,
```

Comment:
Is the cast necessary? I thought TypeScript was smarter than that. We might also want to just use `HTMLElement`.

Reply:
You're right; TypeScript does that for you.